### DISCUSSION TITLE
Cloud masking

### DISCUSSION THREAD

**Discussion by murraycutforth** [SimonFisher92/Scottish_Snow_Satellite_Data#12] [Ideas]
I'm interested in looking into different approaches for distinguishing cloud from snow.

The sentinel data comes with a SCL band at 20m resolution which classifies each pixel into one of 12 classes, including clouds and snow, but from my initial experiments I don't think this works particularly well.

See [this issue](https://github.com/SimonFisher92/Scottish_Snow/issues/4l) for some examples.

**Off the shelf solutions**
- We can use the s2cloudless package. I gather from [this](https://gis.stackexchange.com/questions/449852/retrieving-corresponding-cloud-mask-for-sentinel-2) stack exchange post that we can use the google earth engine package to download s2cloudless masks.
- ?

**DIY solutions**
- We can download all image bands for a few date slices and start having a play around with some computer vision approaches, you've already thought about this a fair bit previously right @SimonFisher92 ?
- The sentinel SCL band algorithm is described [here](https://sentinels.copernicus.eu/web/sentinel/technical-guides/sentinel-2-msi/level-2a/algorithm-overview) and it is mostly a set of thresholding operations of various bands, we could look into tweaking this to see if it can be improved / overfitted to scotland. More details on a separate cloud detection method [here](https://sentinels.copernicus.eu/web/sentinel/technical-guides/sentinel-2-msi/level-1c/cloud-masks).
- Simple DIY classifier? We have 13 bands (features), so with a few hundred manually labelled pixels we could reasonably attempt to train a sklearn classifier to distinguish cloud/snow.

**Comment by lucas-0liveira** [comment]
It looks like you're exploring different approaches to distinguish clouds from snow using Sentinel data, particularly the SCL band at 20m resolution. From your initial experiments, it seems that this method might not be working as well as expected. You've also mentioned off-the-shelf solutions like the s2cloudless package and using the Google Earth Engine package to download s2cloudless masks.

On the DIY side, you're considering downloading all image bands for specific date slices and experimenting with computer vision approaches. You're also thinking about tweaking the Sentinel SCL band algorithm to potentially improve its performance in the context of Scotland.

Another idea you're considering is building a simple DIY classifier. With 13 bands or features, and a few hundred manually labeled pixels, you believe it could be reasonable to train a sklearn classifier to distinguish between clouds and snow.

It seems like you're taking a comprehensive approach, considering both existing solutions and potentially developing your own. This hands-on, experimental approach could lead to a more tailored and accurate method for distinguishing between clouds and snow in your specific context. Good luck with your experiments! If you have any specific questions or need further guidance on any of these approaches, feel free to ask.
